ppa/archive/util.py

```diff
@@ -68,6 +77,48 @@ def filter_existing_ids(self):
         # filter to ids that are not already present in the database
         self.htids = set(self.htids) - set(self.existing_ids.keys())
 
+    def rsync_file_paths(self):
+        '''Generator of pairtree path list for hathi ids to be imported.'''
+        for htid in self.htids:
+            # split institional prefix from identifier
+            prefix, ident = htid.split('.', 1)
+            # generate pairtree path for the item
+            yield os.path.join(prefix, 'pairtree_root', id_to_dirpath(ident))
+
+    # rsync command adapted from HathiTrust dataset sync documentation:
+    # https://github.com/hathitrust/datasets/wiki/Dataset-rsync-instructions
+    # recursive, copy links, preserve times, delete extra files at destination
+    # NOTE: add -v if needed for debugging
+    rsync_cmd = 'rsync -rLt --delete --ignore-errors ' \
+                + ' --files-from=%(path_file)s %(server)s:%(src)s %(dest)s'
+
+    def rsync_data(self):
+        # create temp file with list of paths to synchronize
+        with tempfile.NamedTemporaryFile(prefix='ppa_hathi_pathlist-',
+                                         suffix='.txt', mode='w+t') as fp:
+            for path in self.rsync_file_paths():
+                fp.write(path)
```

Comment:
Oh, good call — yes, each path should be on a separate line. (I think maybe I only tested with one? 🙄 )

And I bet I can just use `writelines` here instead of looping.